chain/messagepool/messagepool.go

```diff
@@ -139,8 +139,24 @@ type MessagePool struct {
 
 	keyCache *lru.Cache[address.Address, address.Address]
 
-	curTsLk sync.RWMutex // DO NOT LOCK INSIDE lk
-	curTs   *types.TipSet
+	// curTs stands for 'current TipSet'
+	curTs *types.TipSet
+
+	// The transactionLk lock is used to ensure consistency when transacting between the TipSet (TS) and the Message Pool.
+	// Any modification to curTs must be wrapped in a transactionLk.
+	// This lock is not needed if you're only interested in a consistent view of the state without changing it.
+	transactionLk sync.RWMutex
+
+	// The stateLk lock is used when you need to interact with the state within the Message Pool.
+	// If the state is being modified, use the Write form of this mutex.
+	// Ensure that the transactionLk lock is acquired before the stateLk lock to maintain consistency.
+	// Example usage:
+	// mp.transactionLk.Lock()
+	// defer mp.transactionLk.Unlock()
+	//
+	// mp.stateLk.Lock()
+	// defer mp.stateLk.Unlock()
+	stateLk sync.RWMutex
```

Comment:
It would simplify the code a lot IMO if we could merge the transactionLk/stateLk into one lock or lock object, is that something that you considered?

Reply:
@snissn What was the resolution to this?

Reply:
I think this was commented on before @fridrik01 got further into reading the PR, and there was a subsequent comment from him clarifying. It is necessary to have two locks for this PR.